test/test-app.js

```diff
@@ -93,6 +80,104 @@ describe('generator-alfresco:app', function () {
       ]);
     });
+    describe('generate model without modules', function () {
+      var osTempDir = path.join(os.tmpdir(), 'temp-test');
+      before(function (done) {
+        helpers.run(path.join(__dirname, '../generators/model'))
+          // generator will create a temp directory and make sure it's empty
+          .inTmpDir(function () {
+            // we want our test to run inside the previously generated directory
+            // and we don't want it to be empty, so this is a hack for that.
+            // process.chdir(path.join(this.osTempDir, 'temp-test'));
+            process.chdir(osTempDir);
+          })
+          .withOptions({
+            'model-name': 'testModel',
+            'model-description': 'test desc',
+            'model-author': 'test author',
+            'model-version': '1.0',
+            'namespace-prefix': 'zz',
+          })
+          .on('end', done);
+      });
+
+      it('model files should not be generated', function () {
+        var modelFile = path.join(osTempDir, 'repo-amp/src/main/amp/config/alfresco/module/repo-amp/model/generated/testModel.xml');
+        var contextFile = path.join(osTempDir, 'repo-amp/src/main/amp/config/alfresco/module/repo-amp/context/generated/test-model-context.xml');
+        assert.noFile([
+          modelFile,
+          contextFile,
+        ]);
+      });
+    });
+  });
+
+  describe('generate project with source amps', function () {
+    this.timeout(60000);
+    var osTempDir = path.join(os.tmpdir(), 'temp-test');
+
+    before(function (done) {
+      helpers.run(path.join(__dirname, '../generators/app'))
+        .inDir(path.join(os.tmpdir(), './temp-test'))
+        .withOptions({ 'skip-install': true })
+        .withPrompts({
+          removeDefaultSourceAmps: false,
+        })
+        .on('end', done);
+    });
+    describe('generate a module', function () {
+      before(function (done) {
+        helpers.run(path.join(__dirname, '../generators/amp'))
+          // generator will create a temp directory and make sure it's empty
+          .inTmpDir(function () {
+            // HACK: we want our test to run inside the previously generated
+            // directory and we don't want it to be empty, so this is a hack
+            // for that.
+            process.chdir(osTempDir);
+          })
+          .withOptions({
+            'force': true, // tests can't handle conflicts
+            'war': 'both',
+            'project-group-id': 'org.alfresco',
+            'project-artifact-id': 'both-customizations',
+            'project-version': '1.0.0-SNAPSHOT',
+            'remove-default-source-samples': false,
+            'create-parent': false,
+            'parent-name': '',
+            'parent-description': '',
+            'repo-name': '',
+            'repo-description': '',
+          })
+          .on('end', function () {
+            helpers.run(path.join(__dirname, '../generators/model'))
```

Comment:
I will do that.. I initially didnt want to add any 'it' tests  because I thought we handle the same case in test-amps.js. But, actually this is a bit different because 'removeDefaultSourceAmps=false' in this test